docs/rules/no-object-literal-type-assertion.md

```diff
@@ -0,0 +1,28 @@
+# Forbids an object literal to appear in a type assertion expression (no-object-literal-type-assertion)
+
+Always prefer `const x: T = { ... };` to `const x = { ... } as T;`
+
+## Rule Details
+
+Examples of **incorrect** code for this rule.
+
+```ts
+const x = { ... } as T;
+```
+
+Examples of **correct** code for this rule.
+
+```ts
+const x: T = { ... };
```

Comment:
looking at the tests - it's valid to cast to `any` or `unknown`.
Should probably document this here just to be clear.

Reply:
doc updated